tests/test_types.py

```diff
@@ -1654,28 +1654,33 @@ def test_bounds_config_exceptions(fn):
 def test_new_type_success():
     a_type = NewType('a_type', int)
     b_type = NewType('b_type', a_type)
+    c_type = NewType('c_type', List[int])
 
     class Model(BaseModel):
         a: a_type
         b: b_type
+        c: c_type
 
-    m = Model(a=42, b=24)
-    assert m.dict() == {'a': 42, 'b': 24}
+    m = Model(a=42, b=24, c=[1, 2, 3])
+    assert m.dict() == {'a': 42, 'b': 24, 'c': [1, 2, 3]}
```

Comment:
```suggestion
    assert m.dict() == {'a': 42, 'b': 24, 'c': [1, 2, 3]}
    assert repr(Model.__fields__['c']) == ...
```
To confirm that `c` has the correct type. (You could also inspect the field's attributes, but that's a bit more involved and I guess this should work as well)

Reply:
I went with the `repr` version, this indeed seemed easier.